user.rb

```diff
@@ -45,7 +55,7 @@ def initialize(first_name:,
     @first_name = first_name
     @last_name = last_name
     @email = email
-    @@all_users << self
+    User.all=( User.all << self )
     User.write_new_user_to_db
```

Comment:
Having a side effect that writes to the file could be a big performance hit. Every time that a new user is instantiated, Ruby needs to open up a file and write to it (which can get quite expensive). Is this necessary here?

Reply:
Which place are you referring to? I guess that doesn't matter really. The only reason I write to the file after every single change is that I never know when someone is going to close the program. If I could have a method trigger when the program is closed (or right before its's closed), I could just write to the file once at the very end of a user's session, similar to how I only read from the file once, when the program is opened.

I just don't know of a way to do that. Is there one?